docs/fix/restore-nicks.rst

```diff
@@ -0,0 +1,30 @@
+fix/restore-nicks
+=================
+
+.. dfhack-tool::
+    :summary: Workaround for the v50.x bug where Dwarf Fortress occasionally erase Dwarf's nicknames.
+    :tags: fort bugfix units
+
+Units occasionally lose their nicknames, in event such as killing a forgotten beast.
+
+`fix/restore-nicks` will save the nicknames of the units once an in-game day, and restore any removed
+nickname.
+
+.. note::
+    It does not make the difference between a nickname removed by the player or by a bug, so
+    it will also prevent to remove the nickname of a dwarf.
+
+Usage
+-----
+
+``enable fix/restore-nicks``
+    Enable saving and restoring of the nicknames daily.
```

Comment:
```suggestion
    Enable daily restoring and saving of the nicknames.
```

Reply:
I know in the code you actually save first, and it looks like correct behavior, but if you say saving first here it might  imply that the data is overwritten before being read, losing the nickname history. This isn't true, but it could be confusing for players.

Reply:
Yep, I can see how it could be confusing